### ISSUE TITLE
Emojis should be wrapped in <span>, have role="img", and have an accessible description with aria-label or aria-labelledby.

### ISSUE THREAD

**Issue by best-lucky1030** [jsx-eslint/eslint-plugin-jsx-a11y#826]
When I use this package in React code, I have one problem.
`Emojis should be wrapped in <span>, have role="img", and have an accessible description with aria-label or aria-labelledby.`

Code is like this:

`<span role="img">❤️</span>`

Searching google, it requires `aria-label or aria-labelledby`
In my case, which do I have to use, `aria-label` or aria-labelledby`?

**Comment by mkrtchian** [comment]
It seems that the `accessible-emoji` rule you are talking about [has been fully deprecated on this package](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/pull/782). Maybe that PR was not integrated at the time the version you are using got released.

As for `aria-label` and `aria-labelledby`, here is the usage:
```html
<span role="img" aria-label="Love">❤️</span>
```

```html
<h1 id="love">Love</h1>
<span role="img" aria-labelledby="love">❤️</span>
```

As you can see `aria-label` is like and `alt` on an `img` tag, it has to be set directly on the tag that needs the text. Whereas `aria-labelledby` takes the id of another tag that is already available on the page, so it's a way to reuse an existing text.
You can choose the one that fits your needs.

**Comment by jessebeach** [comment]
`accessible-emoji` has been deprecated, yes. Thank you @mkrtchian for the assist here.

**Comment by YutaMoriJP** [comment]
@jessebeach 

Is there a workaround for a reusable component like the one below? I would still get linting errors for `Apple`.

```jsx
const Emoji = ({ label, children }: { children: string; label: string }) => (
  <span role="img" aria-label={label}>
    {children}
  </span>
);

// Emojis should be wrapped in <span>, have role="img", and have an accessible description with aria-label or aria-labelledby.

cosnt Apple = () => <Emoji label="Apple">🍎</Emoji>
```

**Comment by jessebeach** [comment]
I think you have 2 options here.

1. If you're consistently using a custom component for emojis in your project, you can turn this rule off. It's going to be deprecated anyway, so that's an OK path forward.
2. Pass the emoji in a prop, rather than children. The rule is only looking at children values, so this will bypass the check: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/e411ce35cfa58181d375544ba5204c35db83678c/src/rules/accessible-emoji.js#L34

```
cosnt Apple = () => <Emoji label="Apple" emoji="🍎" />
```

**Comment by YutaMoriJP** [comment]
@jessebeach Fantastic! Thank you so much 🙏
